pandas/core/arrays/datetimelike.py

```diff
@@ -613,6 +613,18 @@ def _validate_listlike(self, value, allow_object: bool = False):
             # We treat empty list as our own dtype.
             return type(self)._from_sequence([], dtype=self.dtype)
 
+        if hasattr(value, "dtype") and value.dtype == object:
+            # `array` below won't do inference if value is an Index or Series.
+            #  so do so here.  in the Index case, inferred_type may be cached.
```

Comment:
why would't you extract and just pass to `_from_sequence` rather than infer?

Reply:
ive been thinking along similar lines, but cant implement until we get _from_sequence to be appropriately strict